### PR TITLE
Doi refactoring

### PR DESCRIPTION
**Pull request by daisieh**
I refactored the code in DOIIdentifierProvider so that the manipulations of dois and metadata happen once in centralized locations. I also added public accessor methods for retrieving dois and versioned dois.

To test:
make package with two files
approve
make version
delete file
note that there is now one file
approve the one-file version
note that there is one file in version .2
make another version
note that there is still one file
add a file
approve the two-file version (might need to redo the search for unclaimed items at workflow overview)
note that there are two files in version .3

### CODE REVIEW

**Comment by daisieh** [comment]
Argh well I didn't notice that I had accidentally committed the change from #866 here as well as in its own branch. I think it'll all work out okay though: I will delete the other PR.

File: dspace/modules/api/src/main/java/org/dspace/identifier/DOIIdentifierProvider.java [hunk 1]
```diff
@@ -258,67 +250,77 @@ private String mintAndRegister(Context context, Item item, boolean register) thr
...
+                        }
+                    }
+                } else {
+                    previousDOI = getVersionedDataPackageDOIString(previousDOI,1);
+                }
+                updateItemDOIMetadata(previousItem, previousDOI);
+                DOI firstDOI = new DOI(previousDOI, previousItem);
                 if(DryadDOIRegistrationHelper.isDataPackageInPublicationBlackout(item)) {
-                    mint(firstDOI, "http://datadryad.org/publicationBlackout", register, createListMetadata(previous.getItem()));
+                    mint(firstDOI, "http://datadryad.org/publicationBlackout", register, createListMetadata(previousItem));
```
Comment (ghost) [inline dspace/modules/api/src/main/java/org/dspace/identifier/DOIIdentifierProvider.java]: The "http://datadryad.org/publicationBlackout" string seems like it belongs in a properties file.

**Comment by ghost** [comment]
#### Code Review Checklist
- [x] Read through the code to ensure it is:
- [x] Legible
- [x] Well-commented
- [x] Free of any "code smells"
- [ ] Ensure the associated documentation is up to date.
- [x] Compile it on your local machine and test the new functionality.
- [x] Verify that the change is accompanied by adequate testing code. Verify that the code passes the tests.

This tested out as described above on my local VM. I added one comment after reviewing the source files, but it not a blocking issue.

**Comment by daisieh** [comment]
I fixed the property (and removed a confusing but unused private method) and am testing it on my VM to make sure everything still works.

**Comment by daisieh** [comment]
During the checking of the last fix, I found another bug I introduced some time ago with a bad log statement.

**Comment by daisieh** [comment]
Okay @rnathanday, back to you: I think everything works and checks out from my end.

**Comment by ghost** [comment]
Tests out with the updates -- merging

**Pull request merged**